restapi.go

```diff
@@ -2552,3 +2562,234 @@ func (s *Session) FollowupMessageEdit(appID string, interaction *Interaction, me
+	body, err := s.RequestWithBucketID("GET", EndpointChannelActiveThreads(channelID), nil, EndpointThreadMembers(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// PublicArchivedThreads returns archived threads in the channel that are public
+// GET /channels/{channel.id}/threads/archived/public
+//
+// before: returns threads before this timestamp
+// limit: optional maximum number of threads to return
+func (s *Session) PublicArchivedThreads(channelID string, before *time.Time, limit int) (st *ThreadsResponseBody, err error) {
+	uri := EndpointPublicArchivedThreads(channelID)
+
+	v := url.Values{}
+	if limit > 0 {
+		v.Set("limit", strconv.Itoa(limit))
+	}
+
+	if before != nil {
+		v.Set("before", before.Format(time.RFC3339))
+	}
+
+	if len(v) > 0 {
+		uri += "?" + v.Encode()
+	}
+
+	body, err := s.RequestWithBucketID("GET", uri, nil, EndpointPublicArchivedThreads(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// PrivateArchivedThreads returns archived threads in the channel that are private
+// GET /channels/{channel.id}/threads/archived/private
+//
+// before: returns threads before this timestamp
+// limit: optional maximum number of threads to return
+func (s *Session) PrivateArchivedThreads(channelID string, before *time.Time, limit int) (st *ThreadsResponseBody, err error) {
+	uri := EndpointPrivateArchivedThreads(channelID)
+
+	v := url.Values{}
+	if limit > 0 {
+		v.Set("limit", strconv.Itoa(limit))
+	}
+
+	if before != nil {
+		v.Set("before", before.Format(time.RFC3339))
+	}
+
+	if len(v) > 0 {
+		uri += "?" + v.Encode()
+	}
+
+	body, err := s.RequestWithBucketID("GET", uri, nil, EndpointPrivateArchivedThreads(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// JoinedPrivateArchivedThreads returns archived threads in the channel
+// that are of type GUILD_PRIVATE_THREAD, and the user has joined
+// GET /channels/{channel.id}/users/@me/threads/archived/private
+//
+// before: returns threads before this timestamp
+// limit: optional maximum number of threads to return
+func (s *Session) JoinedPrivateArchivedThreads(channelID string, before *time.Time, limit int) (st *ThreadsResponseBody, err error) {
+	uri := EndpointJoinedPrivateArchivedThreads(channelID)
+
+	v := url.Values{}
+	if limit > 0 {
+		v.Set("limit", strconv.Itoa(limit))
+	}
+
+	if before != nil {
+		v.Set("before", before.Format(time.RFC3339))
+	}
+
+	if len(v) > 0 {
+		uri += "?" + v.Encode()
+	}
+
+	body, err := s.RequestWithBucketID("GET", uri, nil, EndpointJoinedPrivateArchivedThreads(""))
+	if err != nil {
+		return
+	}
+
+	err = unmarshal(body, &st)
+	return
+}
+
+// ThreadEdit edits the given thread
+// threadID  : The ID of a Thread
+// name       : The new name to assign the thread.
+func (s *Session) ThreadEdit(channelID string, name string) (*Channel, error) {
+	return s.ThreadEditComplex(channelID, &ThreadEditData{
+		Name: name,
+	})
+}
```

Comment:
Given that `ThreadEditComplex` clears out all properties not specified, I'm not sure that having this function as is here makes a ton of sense. It seems to me like it would quite quickly lead to unexpected behavior and not really have much use.